modules/lfs/pointer_scanner_nogogit.go

```diff
@@ -23,9 +23,28 @@ import (
 func SearchPointerBlobs(ctx context.Context, repo *git.Repository, pointerChan chan<- PointerBlob, errChan chan<- error) {
 	basePath := repo.Path
 
-	catFileCheckReader, catFileCheckWriter := io.Pipe()
-	shasToBatchReader, shasToBatchWriter := io.Pipe()
-	catFileBatchReader, catFileBatchWriter := io.Pipe()
+	fail := func(err error) {
+		errChan <- err
+		close(pointerChan)
+		close(errChan)
+	}
+
+	pipes, err := git.NewPipePairs(3)
+	if err != nil {
+		fail(err)
+		return
+	}
+
+	closeAll := func(err error) {
+		for _, closer := range pipes {
+			_ = closer.CloseWithError(err)
+		}
+	}
+	defer closeAll(nil)
+
```

Comment:
```suggestion
	var morePipes []*git.*PipePair
	var err error

	oldGit := git.CheckGitVersionAtLeast("2.6.0") != nil
	if oldGit {
		pipes, err = git.NewPipePairs(3+2)
	} else {
		pipes, err = git.NewPipePairs(3)
	}
	
	if err != nil {
		errChan <- err
		close(pointerChan)
		close(errChan)
		return
	}

	defer func() {
		for _, closer := range pipes {
			_ = closer.Close()
		}
	}()

```

Reply:
ps maybe there could also be a `git.ClosePipes(pipes, err)` function to close all pipes, I have seen the `for _, closer := range pipes { _ = closer.Close() }` a few times